spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -20,166 +20,124 @@
+    }
 
-            String urlToGetServiceApi = securityUtil.attachGatewayApiKeyToUrl(
-                    adminBaseUrl + adminResourceToGetServiceByApiKeyValue + apiKeyValue);
+    private Authentication createAdminsAuthToken(String apiKeyValue) {
+        log.debug("Creating admins authentication token using API key with the value {}.", apiKeyValue);
+        Set<SimpleGrantedAuthority> authoritiesGrantedToYggAdmin = new HashSet<>();
+        authoritiesGrantedToYggAdmin.add(new SimpleGrantedAuthority(
+                ServiceApiAccessRightsVoter.ACCESS_SERVICE_AUTHORITY_PREFIX + "refresh"));
+        ApiKeyAuthToken apiKeyAuthToken = new ApiKeyAuthToken(apiKeyValue, authoritiesGrantedToYggAdmin);
+        apiKeyAuthToken.setAuthenticated(true);
+        log.info("Admin is successfully authenticated using the API key with the value {}.", apiKeyValue);
+        return apiKeyAuthToken;
+    }
 
-            ResponseEntity<ServiceApi> responseEntity = restTemplate.getForEntity(urlToGetServiceApi, ServiceApi.class);
+    private Authentication verifyApiKeyAndIfAuthorizedCreateConsumerAuthToken(String apiKeyValue) {
+        log.debug("Verifying the API with the value {} from consumer.", apiKeyValue);
 
-            if (responseEntity != null) {
-                ServiceApi serviceApi = responseEntity.getBody();
-                log.debug("Service API was found by CoatRack admin: " + serviceApi);
+        boolean isApiKeyValid;
+        ApiKey apiKey;
 
-                return serviceApi;
-            } else {
-                log.error("Communication with Admin server failed, result is: " + responseEntity);
-                throw new AuthenticationServiceException("Communication with auth server failed");
-            }
-        } catch (Exception e) {
-            throw new AuthenticationServiceException("An error occured, when trying to get service API data from admin server", e);
+        try {
+            apiKey = apiKeyFetcher.requestApiKeyFromAdmin(apiKeyValue);
+            isApiKeyValid = localApiKeyVerifier.isApiKeyValid(apiKey);
+        } catch (RestClientException e) {
+            log.info("Trying to verify consumers API key with the value {}, the connection to admin failed.",
+                    apiKeyValue);
+            apiKey = localApiKeyManager.findApiKeyFromLocalApiKeyList(apiKeyValue);
+
+            if(apiKey == null)
+                return null;
+            else
+                isApiKeyValid = localApiKeyVerifier.isApiKeyAuthorizedConsideringLocalApiKeyList(apiKeyValue);
         }
+
+        if (isApiKeyValid) {
+            return createConsumersAuthToken(apiKey);
+        } else
+            return null;
     }
 
+    private ApiKeyAuthToken createConsumersAuthToken(ApiKey apiKey) {
+        log.debug("Create consumers authentication token using API key with the value {}.", apiKey.getKeyValue());
+        ServiceApi serviceApi = apiKey.getServiceApi();
```

Comment:
Just nitpicking: do we need this variable? It seems to be used only in the next line.

Reply:
It was not needed and is now removed.